urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/utils/lang/TranslaterFormatter.java

```diff
@@ -0,0 +1,37 @@
+package com.kokakiwi.mclauncher.utils.lang;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TranslaterFormatter
+{
+    public static String format(String str, String[]... keys)
+    {
+        final Map<String, String> keyMap = new HashMap<String, String>();
+        for (final String[] key : keys)
+        {
+            if (key.length == 2)
+            {
+                keyMap.put(key[0], key[1]);
+            }
+        }
+        
+        return format(str, keyMap);
+    }
+    
+    public static String format(String str, Map<String, String> keys)
+    {
+        String result = str;
+        
+        if (keys != null)
+        {
+            for (final String key : keys.keySet())
+            {
+                result = result.replaceAll("\\{" + key.toUpperCase() + "\\}",
+                        keys.get(key));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.